### ISSUE TITLE
node.ssh.put fails : raise SSHException('SSH session not active')

### ISSUE THREAD

**Issue by dlovell** [jtriley/StarCluster#119]
This occurs when I try to put a file to each node in starcluster_plugin.py

https://github.com/mit-probabilistic-computing-project/Cloudless/blob/mrjobify/starcluster_plugin.py

If I immediately rerun with 'starcluster runplugin Cloudless mycluster', everything works fine.

Note that several node.ssh.execute commands prior to this command run fine (which seems odd but I don't know if ssh.execute and ssh.put use different resources)

**Comment by jtriley** [comment]
@dlovell My apologies for the delay. The SSHClient class checks to make sure the transport is active and reconnects it if necessary each time it's requested. Looking at the SCPClient class I can see it's not doing this and I think this is likely the issue - I'll try to implement a fix for this soon.

Now that it's been quite some time, are you running into this often?

**Comment by dlovell** [comment]
Hi Justin,

First, thanks for StarCluster!  I used EMR for a bit (and it was a
tremendous pain) because I didn't know StarCluster had a hadoop plugin but
now can fluidly spin up large clusters.

I changed a few things recently (the types of instances I'm spinning up,
the plugins I'm loading) and it is now (possibly) intermittent rather
than deterministically always happening, but it used to always happen and I
would always run 'starcluster runplugin Cloudless MYCLUSTER'.

I'm unsure, but think it could have been plugins/order of plugins which
caused the issue: I went from 'plugins = ipcluster, Cloudless' to 'plugins
= hadoop, Cloudless'

Also, I its possible it may be deterministically NOT happening with the new
config and that it happening is just the occasions where I load one of the
cluster configurations that still have 'plugins = ipcluster, Cloudless'.  I
can spin up a machine or two  to test if you want.

Thanks Much!
Dan
